Packs/GoogleKubernetesEngine/Integrations/GoogleKubernetesEngine/GoogleKubernetesEngine.py

```diff
@@ -0,0 +1,928 @@
+###########
+# IMPORTS #
+###########
+# STD packages
+from typing import Dict, Callable, Tuple, Any, Optional, List
+from pathlib import Path
+# 3-rd party packages
+from google.cloud.container_v1 import ClusterManagerClient
+from google.protobuf.json_format import MessageToDict
+from google.protobuf.message import Message
+from google.cloud.container_v1 import enums
+# Local packages
+import demistomock as demisto
+from CommonServerPython import *  # noqa: E402 lgtm [py/polluting-import]
+from CommonServerUserPython import *  # noqa: E402 lgtm [py/polluting-import]
+
+#########
+# Notes #
+#########
+"""
+    Development info:
+    1.This integration implements the gke cli of google cloud platform, For more info -
+        https://cloud.google.com/sdk/gcloud/reference/container/clusters
+    2.In the implementation we use the official "google-cloud-container" sdk, For more info -
+        https://googleapis.dev/python/container/latest/gapic/v1/types.html#google.cloud.container_v1
+    3. Authentication done by Google service account, For more info -
+        https://cloud.google.com/iam/docs/service-accounts
+    Notice - Between updates GKE will not allow to perform new update until the last one finished, check the status in
+             operation entry.           
+"""  # noqa W291
+####################
+# GLOBAL CONSTUNTS #
+####################
+INTEGRATION_NAME = 'Google Kubernetes Engine'
+INTEGRATION_COMMAND_NAME = 'gcloud'
+INTEGRATION_CONTEXT_NAME = 'GKE'
+CLUSTER_CONTEXT = f'{INTEGRATION_CONTEXT_NAME}.Cluster(val.Name && val.Name == obj.Name)'
+NODE_POOL_CONTEXT = f'{INTEGRATION_CONTEXT_NAME}.NodePool(val.Name && val.Name == obj.Name)'
+OPERATION_CONTEXT = f'{INTEGRATION_CONTEXT_NAME}.Operation(val.Name && val.Name == obj.Name)'
+OPERATION_TABLE = ['Name', 'Zone', 'Status', 'StartTime']
+COMMAND_OUTPUT = Tuple[str, Dict[str, Any], Dict[str, Any]]
+API_TIMEOUT = 90
+
+
+####################
+# HELPER FUNCTIONS #
+####################
+
+
+def google_client_setup(json_configuration: str) -> ClusterManagerClient:
+    """ Setup client for service acount in google cloud - For more information:
+        https://cloud.google.com/iam/docs/service-accounts
+
+    Args:
+        json_configuration: Json configuration file content from IAM.
+
+    Returns:
+        ClusterManagerClient: client manager.
+    """
+    temp_configuration_file = Path.cwd() / 'credentials_service_account.json'
+    try:
+        # Create temp file to apply the clientt from - doesn't support file like object
+        temp_configuration_file.write_text(json_configuration)
+        # Create client
+        client = ClusterManagerClient.from_service_account_json(filename=temp_configuration_file)
```

Comment:
check if you can avoid the file creation as we do in other Google integrations

Reply:
Done